### PR TITLE
[TH] Add translation of `contribution guide`

### PR DESCRIPTION
**Pull request by earth1168**
## Self-check

- [x] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)
- [x] *(translations only)* The changes are reviewed on GitHub [by a fluent speaker](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#review)

TicClick: `SKIP_WIKILINK_CHECK` because there's still a section link that can't be fixed (no corresponding heading in an outdated article)

### CODE REVIEW

**Comment by TicClick** [comment]
please don't tick the second checkbox until there is a review you applied from another person, posted on GitHub

File: wiki/osu!_wiki/Contribution_guide/th.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+- **[Content parity](/wiki/Article_styling_criteria/Writing#content-parity)** บทแปลจะต้องมีเนื้อหาเดิมกับบทความต้นฉบับ (การเปลี่ยนแปลงทางด้านการเว้นวรรค การใช้คำ หรือการจัดรูปแบบ เป็นสิ่งที่เกิดขึ้นได้) แทนที่จะเปลี่ยนเนื้อหาบทความต้นฉบับให้ลอง [open an issue](https://github.com/ppy/osu-wiki/issues/new) หรือเปิด pull request อันใหม่ที่แก้ไขบทความต้นฉบับ ถ้าคุณรู้สึกว่ามันยังไม่สมบูรณ์ ไม่ถูกต้อง หรือเก่าเกินไป
+- **โครงสร้าง และการจัดรูปแบบ** ลองดูตัวอย่างงานของคุณผ่านเครื่องมืออย่าง [jbt's Markdown Editor](https://jbt.github.io/markdown-editor/) เพื่อทำให้แน่ใจว่างานของคุณมีหน้าตาเหมือนกับที่คุณคิดไว้
+- **รูปภาพทุกรูป และไฟล์อื่น ๆ ที่ไม่ใช้ตัวหนังสือ** จะต้อง [เล็กกว่า 1 megabyte](/wiki/Article_styling_criteria/Formatting#file-size) **ภาพหน้าจอ** จะต้องใช้สกินเริ่มต้นของ osu! และใช้ [การตั้งค่าที่กำหนด](/wiki/Article_styling_criteria/Formatting#screenshots-of-gameplay) โดยมีขนาดไม่เกิน 1280×720
+
+สำหรับมาตราฐานการเขียน และการจัดรูปแบบ สามารถดูได้ที่ [เกณฑ์การจัดรูปแบบบทความ](/wiki/Article_styling_criteria) โดยไม่แนะนำให้อ่านตั้งแต่ต้นจนจบในครั้งเดียว และผู้ตรวจสอบจะแจ้งจุดที่ผิดพลาดและต้องแก้ไขหากพบเจอ
```
Comment (HelloYeew) [inline wiki/osu!_wiki/Contribution_guide/th.md:40]: A bit typo here

```suggestion
สำหรับมาตรฐานการเขียน และการจัดรูปแบบ สามารถดูได้ที่ [เกณฑ์การจัดรูปแบบบทความ](/wiki/Article_styling_criteria) โดยไม่แนะนำให้อ่านตั้งแต่ต้นจนจบในครั้งเดียว และผู้ตรวจสอบจะแจ้งจุดที่ผิดพลาดและต้องแก้ไขหากพบเจอ
```

File: wiki/osu!_wiki/Contribution_guide/th.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+## Pull request
+
+หลังจากมีการตรวจทานการเปลี่ยนแปลงต่าง ๆ แล้ว เลือก committed และ pushed ไปยัง fork ของคุณ โดยคุณต้องแจ้งผู้ดูแลโดยการเปิด pull request
```
Comment (HelloYeew) [inline wiki/osu!_wiki/Contribution_guide/th.md:44]: I think a meaning in Thai compare to English version is a bit wrong here.

```suggestion
หลังจากตรวจทานการเปลี่ยนแปลงต่าง ๆ, commit และ push ไปยัง fork ของคุณแล้ว คุณต้องเสนอมันไปยังผู้ดูแลโดยการเปิด pull request
```

File: wiki/osu!_wiki/Contribution_guide/th.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ในกรณีที่คุณต้องการความช่วยเหลือ หรือคำแนะนำในขั้นตอนไหนก็ตาม อย่าลังเลที่จะถามเข้ามาในช่อง `#osu-wiki` ของ [osu!dev Discord server](/wiki/Community/osu!dev_Discord_server).
+
+## ขอบเขตของงาน
+
+ถ้าคุณอยากจะช่วย แต่ไม่แน่ใจว่าจะเริ่มต้นอย่างไร ลองดู [osu! wiki maintenance § Routines](/wiki/osu!_wiki/Maintenance#routines) เพื่อค้นหารายการของงานที่ต้องการให้ทำโดยทั่วไป และวิธีการในการช่วยเหลือเหล่าผู้มีส่วนร่วมคนอื่น ๆ  
+ในการทำให้ตัวคุณคุ้นเคยกับรูปแบบภาษาที่ใช้ใน wiki ลองดู [Markdown Cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) ของ Adam Pritchard 
+
+## การแก้ไข wiki
+
+สำหรับกระบวนการการมีส่วนร่วมทั่วไป ลองตรวจสอบ [GitHub flow - GitHub Docs](https://docs.github.com/en/get-started/quickstart/github-flow)*
```
Comment (HelloYeew) [inline wiki/osu!_wiki/Contribution_guide/th.md:16]: Missing star here.

```suggestion
*สำหรับกระบวนการการมีส่วนร่วมทั่วไป ลองตรวจสอบ [GitHub flow - GitHub Docs](https://docs.github.com/en/get-started/quickstart/github-flow)*
```

**Review by HelloYeew — CHANGES_REQUESTED** [review]
Looking good overall 👍. I marked some typo or some text that I think it can be better here. Maybe you need to check and fix some check warning that CI has marked it in `Files changed` tab.

File: wiki/osu!_wiki/Contribution_guide/th.md (hunk 1 above)
Comment (earth1168) [reply wiki/osu!_wiki/Contribution_guide/th.md]: Thanks for the review! Sorry for the typo and wrong translation, fixed

File: wiki/osu!_wiki/Contribution_guide/th.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+1. ใน **Fork ของคุณ** ของ osu! wiki หา droupdown ที่เขียนว่า `master↓` และเลือก branch ที่มีการเปลี่ยนแปลงของคุณ
+2. เลือกปุ่ม `Contribute↓` และเลือก `Open pull request`
+3. เติมรายละเอียดตาม [Best practices § Opening a pull request](/wiki/osu!_wiki/Contribution_guide/Best_practices#opening-a-pull-request) และเลือก `Create pull request`
+
+## การ Review
+
+การเปลี่ยนแปลงทุกอย่างใน osu! wiki จะได้รับการกลั่นกรอง ในระหว่างขั้นตอนการตรวจสอบ จะมีผู้ประสานงานชี้จุดที่ผิดพลาด และวิธีการในการแก้ไขในรูปแบบ free form หรือบอกตรง ๆ ใน pull request เช่นเดียวกับการตรวจสอบแบบอื่น ๆ คุณต้องมีการเขียนความคิดเห็นเพิ่มเติมโดยการ [ตั้งข้อเสนอแนะ](/wiki/osu!_wiki/Contribution_guide/Best_practices#applying-reviews) หรืออธิบายเพิ่มเติมว่าทำไมถึงอยากจะให้จุดเหล่านั้นเป็นไปตามที่คุณแก้ไข
+
+ถ้าผ่านไปหลายวันแล้วไม่มีใครมาตรวจสอบ ลองทำตามขั้นตอนเหล่านี้
```
Comment (agatemosu) [inline wiki/osu!_wiki/Contribution_guide/th.md:53]:
```suggestion
ถ้าผ่านไปหลายวันแล้วไม่มีใครมาตรวจสอบ ลองทำตามขั้นตอนเหล่านี้

```

File: wiki/osu!_wiki/Contribution_guide/th.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+- ตรวจให้แน่ใจว่าคุณได้แก้ไขตามคำแนะนำทั้งหมดแล้ว บางทีผู้ตรวจสอบอาจรอคำตอบของคุณอยู่
+- ลองถามผู้เขียน osu! wiki คนอื่น ๆ ใน [osu!dev Discord server](/wiki/Community/osu!dev_Discord_server) (ช่อง `#osu-wiki` ) หรือใน GitHub comment
+- ในการขอความช่วยเหลือด้านการแปล ตรวจสอบ pull request ที่ merged เรียบร้อยแล้วของภาษาของคุณ ลองเข้าไปติดต่อผู้ตรวจสอบ หรือผู้เขียนของ pull request อันนั้น ([example GitHub query](https://github.com/ppy/osu-wiki/pulls?q=is%3Apr+is%3Amerged+%5BID%5D))
+- ลองให้เพื่อนมาดูแบบเร็ว ๆ
```
Comment (agatemosu) [inline wiki/osu!_wiki/Contribution_guide/th.md:57]:
```suggestion
ถ้าผ่านไปหลายวันแล้วไม่มีใครมาตรวจสอบ ลองทำตามขั้นตอนเหล่านี้

- ตรวจให้แน่ใจว่าคุณได้แก้ไขตามคำแนะนำทั้งหมดแล้ว บางทีผู้ตรวจสอบอาจรอคำตอบของคุณอยู่
- ลองถามผู้เขียน osu! wiki คนอื่น ๆ ใน [osu!dev Discord server](/wiki/Community/osu!dev_Discord_server) (ช่อง `#osu-wiki` ) หรือใน GitHub comment
- ในการขอความช่วยเหลือด้านการแปล ตรวจสอบ pull request ที่ merged เรียบร้อยแล้วของภาษาของคุณ ลองเข้าไปติดต่อผู้ตรวจสอบ หรือผู้เขียนของ pull request อันนั้น ([example GitHub query](https://github.com/ppy/osu-wiki/pulls?q=is%3Apr+is%3Amerged+%5BID%5D))
- ลองให้เพื่อนมาดูแบบเร็ว ๆ
```

Comment (earth1168) [reply wiki/osu!_wiki/Contribution_guide/th.md]: Thanks for your help!

**Review by TicClick — APPROVED** [review]

**Pull request merged**